webroot/js/modules/Panels/CachePanel.js

```diff
@@ -0,0 +1,35 @@
+export default (($) => {
+  const addMessage = (text) => {
+    $('<p>' + text +'</p>')
+      .appendTo('.c-cache-panel__messages')
+      .fadeOut(2000);
+  };
+
+  const init = () => {
+    $(document).on('click', '.js-clear-cache', function triggerCacheClear(e) {
+      const el = $(this);
+      const name = el.attr('data-name');
+      const baseUrl = el.attr('data-url');
+      const csrf = el.attr('data-csrf');
+
+      $.ajax({
+        headers: { 'X-CSRF-TOKEN': csrf },
+        url: baseUrl,
+        data: { name },
+        dataType: 'json',
+        type: 'POST',
+        success(data) {
+          addMessage(data.message);
+        },
+        error(jqXHR, textStatus, errorThrown) {
+          addMessage(errorThrown);
+        },
+      });
+      e.preventDefault();
+    });
+  };
+
+  return {
+    init,
+  };
```

Comment:
Should this use the `onEvent` pattern you used in Mail and History panels?

Reply:
I usually prefer to use the `$(document).on('click', '.js-some-class', function(){}` because this only has to be executed once and it works even with content that gets added via AJAX. But if this gets excecuted multiple times then we have mutliple event listeners on the same element which in this case would trigger multiple ajax requests to clear the cache.

But since we now have a pretty clear initialization event for each panel this is not really necessary and therefore I can change that back to `$('.js-some-class').on('click', function(){}` so the event listeneres get re-added everytime a panel is reloaded.